spec/controllers/works/default_rails_actions_spec.rb

```diff
@@ -277,30 +326,55 @@ def call_with_params(params)
   end
 
   describe "index" do
-    before do
-      @fandom = create(:canonical_fandom)
-      @work = create(:work, fandom_string: @fandom.name)
-    end
+    let(:fandom) { create(:canonical_fandom) }
+    let!(:work) { create(:work, fandom_string: fandom.name) }
 
     it "returns the work" do
       get :index
-      expect(assigns(:works)).to include(@work)
+      expect(assigns(:works)).to include(work)
     end
 
     it "sets the fandom when given a fandom id" do
-      params = { fandom_id: @fandom.id }
+      params = { fandom_id: fandom.id }
       get :index, params: params
-      expect(assigns(:fandom)).to eq(@fandom)
+      expect(assigns(:fandom)).to eq(fandom)
+    end
+
+    it "redirects to tag page for noncanonical tags" do
+      noncanonical_tag = create(:character)
+      get :index, params: { id: work, tag_id: noncanonical_tag.name }
+      expect(response).to redirect_to(tag_path(noncanonical_tag))
+    end
+
+    it "redirects to tags works page for noncanonical merged tags page" do
```

Comment:
Should this example and the one on line 357 be tested under the `without caching` section, since we already have the same tests in the _with_ caching section?

https://github.com/otwcode/otwarchive/blob/584c4daabf3caf755692f9202256bd9af4a4ec78/spec/controllers/works/default_rails_actions_spec.rb#L456-L474

It would also makes sense to use the same descriptions for the new tests that we used for the existing tests. That will make it clearer the examples just cover the same thing in different contexts. (If you could change the existing `describe "with caching"` and `describe "without caching"` to _context_ instead, that would be great.)

Reply:
Yes, fixed